build-tools/packages/build-cli/src/flags.ts

```diff
@@ -253,6 +253,21 @@ export const selectionFlags = {
 	}),
 };
 
+/**
+ * This interface is used for type enforcement of selection flags. The default oclif typing is complex and difficult to
+ * simplify, so this type just mirrors the object above with simpler typing. This should ONLY be used when processing
+ * raw flags.
+ *
+ * @internal
+ */
+export interface selectionFlags {
```

Comment:
I suspect all these fields can be made readonly.

Reply:
also the arrays inside can also prabably be readonly.

Reply:
Good point. I'll update these.